src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs

```diff
@@ -91,11 +91,23 @@ protected override async Task<object> InvokeCore(object[] parameters, FunctionIn
             };
 
             string invocationId = context.ExecutionContext.InvocationId.ToString();
-            _logger.LogTrace($"Sending invocation id:{invocationId}");
-            await _functionDispatcher.InvokeAsync(invocationContext);
-            var result = await invocationContext.ResultSource.Task;
+            ScriptInvocationResult result = new();
+
+            try
+            {
+                _logger.LogTrace("Sending invocation id: '{id}", invocationId);
+
+                await _functionDispatcher.InvokeAsync(invocationContext);
+                result = await invocationContext.ResultSource.Task;
+
+                await BindOutputsAsync(triggerValue, context.Binder, result);
+            }
+            catch (Exception ex) when (invocationContext.ResultSource.Task.IsCanceled)
```

Comment:
Any reason we don't use `catch (OperationCanceledException ex)`? Can you comment why we do it that way?

If we are looking for handling only if a _specific_ `await` statement above is canceled, should we wrap it with its own `try/catch (OperationCanceledException ex)` statement? It might be a bit clearer to the reader what our intentions are then.

Reply:
I want to make sure we're only catching this when we cancel the ResultSource. This only happens at one point in our code base (GrpcWorkerChancel -> InvocationRequest). I'm trying to avoid messing with any other flow we have here.

Yup it is only for a specific await statement so I can update the code for the try/catch. I think the exception thrown in TaskCanceledException and I don't know if something else might cause it or not so I feel like having the resultsource check doesn't hurt?